dashboard/test/models/user_level_test.rb

```diff
@@ -75,74 +118,97 @@ class UserLevelTest < ActiveSupport::TestCase
-    assert_equal 0, UserLevel.perfect.count
+    assert_difference 'UserLevel.count' do
+      assert_no_difference 'UserLevel.attempted.count' do
+        assert_no_difference 'UserLevel.passing.count' do
+          assert_no_difference 'UserLevel.perfect.count' do
+            UserLevel.create(
+              user: @user,
+              level: @level,
+              attempts: 0,
+              best_result: nil
+            )
+          end
+        end
+      end
+    end
   end
 
   test "attempted, passed, and perfected scopes for attempted result" do
-    UserLevel.create(user: @user, level: @level, attempts: 0, best_result: Activity::MINIMUM_FINISHED_RESULT)
-
-    assert_equal 1, UserLevel.count
-    assert_equal 1, UserLevel.attempted.count
-    assert_equal 0, UserLevel.passing.count
-    assert_equal 0, UserLevel.perfect.count
+    assert_difference 'UserLevel.count' do
+      assert_difference 'UserLevel.attempted.count' do
+        assert_no_difference 'UserLevel.passing.count' do
+          assert_no_difference 'UserLevel.perfect.count' do
+            UserLevel.create(
+              user: @user,
+              level: @level,
+              attempts: 0,
+              best_result: Activity::MINIMUM_FINISHED_RESULT
+            )
+          end
+        end
+      end
+    end
   end
 
   test "attempted, passed, and perfected scopes for passed result" do
-    UserLevel.create(user: @user, level: @level, attempts: 0, best_result: Activity::MINIMUM_PASS_RESULT)
-
-    assert_equal 1, UserLevel.count
-    assert_equal 1, UserLevel.attempted.count
-    assert_equal 1, UserLevel.passing.count
-    assert_equal 0, UserLevel.perfect.count
+    assert_difference 'UserLevel.count' do
+      assert_difference 'UserLevel.attempted.count' do
+        assert_difference 'UserLevel.passing.count' do
+          assert_no_difference 'UserLevel.perfect.count' do
+            UserLevel.create(
+              user: @user,
+              level: @level,
+              attempts: 0,
+              best_result: Activity::MINIMUM_PASS_RESULT
+            )
+          end
+        end
+      end
+    end
   end
 
   test "attempted, passed, and perfected scopes for perfected result" do
-    UserLevel.create(user: @user, level: @level, attempts: 0, best_result: Activity::MAXIMUM_NONOPTIMAL_RESULT + 1)
-
-    assert_equal 1, UserLevel.count
-    assert_equal 1, UserLevel.attempted.count
-    assert_equal 1, UserLevel.passing.count
-    assert_equal 1, UserLevel.perfect.count
+    assert_difference 'UserLevel.count' do
+      assert_difference 'UserLevel.attempted.count' do
+        assert_difference 'UserLevel.passing.count' do
+          assert_difference 'UserLevel.perfect.count' do
+            UserLevel.create(
+              user: @user,
+              level: @level,
+              attempts: 0,
+              best_result: Activity::MAXIMUM_NONOPTIMAL_RESULT + 1
+            )
+          end
+        end
+      end
+    end
```

Comment:
This and similar above were necessitated as a result of the brittleness of the existing tests, specifically the creation of additional `user_levels` in the `setup`.